composer/trainer/trainer.py

```diff
@@ -2116,22 +2107,21 @@ def fit(
             )
 
             for evaluator in evaluators:
-                validate_eval_automicrobatching(evaluator.auto_microbatching, self.state.device)
-                if evaluator.auto_microbatching and hasattr(evaluator.dataloader, 'seq_parallel_world_size'):
-                    raise ValueError('`validate_eval_automicrobatching` is not compatible with sequence parallelism.')
-                if isinstance(evaluator.dataloader.get_num_samples_in_batch, int) and hasattr(
-                    evaluator.dataloader,
-                    'seq_parallel_world_size',
-                ) and evaluator.dataloader.get_num_samples_in_batch * evaluator.dataloader.seq_parallel_world_size != 1:  # type: ignore
-                    raise ValueError(
-                        '`Sequence parallelism requires a microbatch size of 1 distributed over the sequence parallel group.',
-                    )
+                validate_evaluator(evaluator, self.state.device)
 
             if len(evaluators) == 0:
                 if eval_subset_num_batches != -1:
-                    raise ValueError('Specifying `eval_subset_num_batches` without an `eval_dataloader` has no effect.')
-                if eval_interval != 1:
-                    raise ValueError('Specifying `eval_interval` without an `eval_dataloader` has no effect.')
+                    warnings.warn(
+                        f'Specifying `eval_subset_num_batches={eval_subset_num_batches}` without an `eval_dataloader` '
+                        'has no effect. If trying to run an evaluator, make sure `eval_dataloader` is specified. '
+                        'Otherwise, set `eval_subset_num_batches` to default value -1.',
+                    )
+                if eval_interval != 0 and eval_interval != 1:
+                    warnings.warn(
```

Comment:
Will this raise multiple warnings?

Reply:
It should raise multiple warnings as the two args are different? Do you want to batch?

Reply:
Ah gotcha. Probably easier to keep as is, batching would just complicate the code